moo.js

```diff
@@ -380,9 +404,10 @@
   }
```

Comment:
You still need to copy `error` in `Lexer#clone`. Actually, using `assign({}, s, {regexp: …})` might be better so we don't have to keep updating it.